### PR TITLE
[Offline area] Refactor code related to loading local/remote tile overlays

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes https://github.com/google/ground-android/issues/1790

Things done:
 * Convert `TileSource` enum to sealed data class
 * Simplify `OfflineAreaRepository`
 * Replace `List<TileSource>` with `TileSource` when adding overlay for simplicity. If we need to support multiple tile sources, then we can change it back later.

This is a pure refactor and doesn't change app behavior.

Verified for regressions by testing that remote mog overlay is loaded in offline area screen and locally downloaded tiles are loaded on home screen.

@gino-m  PTAL?

### CODE REVIEW

**Review by gino-m — CHANGES_REQUESTED** [review]
One question, LGTM

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 1]
```diff
@@ -115,11 +115,11 @@ constructor(
   var currentCameraPosition = MutableStateFlow<CameraPosition?>(null)
     private set
 
-  val offlineTileSources: LiveData<List<TileSource>> =
+  val offlineTileSources: LiveData<TileSource?> =
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:118]: While refactoring the code, I found that not using a list of tile sources was the key to simplifying the codebase. If we need to support multiple tile sources at once someday, we should add support for it then.

[the remark this answers is not in the document]
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:118]: Sgtm!

**Review by gino-m — APPROVED** [review]

**Pull request merged**